src/errors.rs

```diff
@@ -23,6 +23,8 @@ pub enum InternalError {
     NoSquareRoots,
     /// Elements are not coprime
     NotCoprime,
+    /// One or more of the integer inputs to the Chinese remainder theorem were outside the expected range.
+    InvalidIntegers,
```

Comment:
question: I added this to our big old `InternalError`, but I could alternately take it and `NotCoprime` (and maybe others) and start a `PiModError` enum in that module. This is a little jumpstart on our existing error handling issues.

Reply:
Good question. I will add some thoughts here about error handling. Tagging @indomitableSwan @tyurek for visibility!

Fundamentally, our error handling should be useful:
- It should be easy to use, this usually translates to just being able to use `?` and having it just work. Adding `From` trait implementations between types can help avoid `.map_error(...)`, etc. We shouldn't be constantly battling the `?` operator or too many error type conversions.
- It should provide useful information. This can be in the form of more granular variants for our error type, where the name carries enough meaning and is granular enough to differentiate it from other errors. We could also choose to have our error type carry data, e.g. `InvalidInteger` vs `InvalidInteger(BigNumber)` which would carry the offending number as part of its data.
- Error origination: When you have errors originating from several layers down the call stack, and getting propagated up it can be difficult to tell exactly where the error originated from. For lock-keeper, logging has largely made this a non-issue. I expect we can do the same here.

Other things to think about:
- Matching on variants: Will we (or an application using our library) ever need to match on the error variant type and branch control flow based on the result? If so, we should keep this in mind when designing our variants.

This may sound complicated, but in practice it shouldn't be too bad. Just wanted to list things to keep in mind. We also want to avoid over-engineering, e.g. what would be the issue with having one big enum with all our errors? I don't really see this causing issues.

Reply:
I tried separating out a `PiModError` but it failed on the first point, primarily. Ugly `Result` types and kind of funky returning. So, I decided to keep this and other `PiMod` exclusive errors here.